_posts/2020-07-12-ray-tune.md

```diff
@@ -0,0 +1,264 @@
+
+This example is almost identical to [one of Ray's examples][sgd-eg] and [one of
+Scikit-learn's examples][skl-eg]. Clearly, Dask-ML offers a Scikit-learn
+compatible model selection API.
+
+[skl-eg]:https://scikit-learn.org/stable/auto_examples/model_selection/plot_randomized_search.html#sphx-glr-auto-examples-model-selection-plot-randomized-search-py
+[sgd-eg]:https://github.com/ray-project/tune-sklearn/blob/31f228e21ef632a89a74947252d8ad5323cbd043/examples/sgd.py
+
+> 2\. **Modern hyperparameter tuning techniques:** tune-sklearn is the only
+> Scikit-Learn interface that allows you to easily leverage Bayesian
+> Optimization, HyperBand, and other optimization techniques by simply toggling
+> a few parameters.
+
+Dask-ML also offers modern hyperparameter tuning techniques. Hyperband is the
+most prominent example; after the introduction in 2016, [the paper by Li et.
+al][hyperband-paper] has been cited [over 470 times][470] and has been
+implemented in many different libraries including [Dask-ML][hscv],
+[keras-tune], [Optuna], AutoML[^automl] and [Microsoft's NNI][nni].
+
+Hyperband is so popular because it's so effective. In the introduction paper by
+Li et.  al, they showed a rather drastic improvement when comparing with all
+the relevant implementations,[^hyperband-figs] and this drastic improvement has
+remained in follow-up works.[^follow-up]
+
+
+[^follow-up]:See Figure 1 of [the BOHB paper][BOHB paper] and [a paper][blippar] from an augmented reality company.
+
+[^hyperband-figs]:See Figures 4, 7 and 8 in "[Hyperband: A Novel Bandit-Based Approach toHyperparameter Optimization][hyperband-paper]."
+
+[blippar]:https://arxiv.org/pdf/1801.01596.pdf
+[BOHB paper]:http://proceedings.mlr.press/v80/falkner18a/falkner18a.pdf
+[nni]:https://nni.readthedocs.io/en/latest/Tuner/HyperbandAdvisor.html
+
+Even Bayesian hyperparameter optimization have been adapted to the Hyperband
+implementation in AutoML's [BOHB]. Dask-ML could use more Bayesian sampling
+approaches; however, this is difficult given the distributed nature of Dask and
+the serial nature of Bayesian sampling algorithms. Bayesian sampling treats
+models as black boxes, and tries to deduce the set of inputs that maximize the
+black box's quality. If there were an infinite number of workers, the Bayesian
+sampling algorithm wouldn't have time to deduce anything before initializing
+models. Regardless, [BOHB] has an intelligent way of scaling to distributed
+systems.[^scaling]
+
+[^scaling]:As mentioned in Section 4.2 and illustrated in Figure 2 of [the BOHB paper][BOHB paper].
+
+The additional features of Dask-ML's hyperparameter optimization are also
+modern, especially in the context of natural language processing (NLP).
+[Avoiding repeated work] in Scikit-learn Pipelines is a huge feature when
+performing NLP because the initial stages of analysis take an excessive amount
+of computation and memory.[^openai] It's future work to combine this with Dask-ML's
+`HyperbandSearchCV`.
```

Comment:
Is this future work the need for a `Pipeline.partial_fit`? If so, perhaps link to that issue / PR (showing that we're not just taking the scikit-learn API as gospel, but are working with the community to improve it).

Reply:
I added a note about how we added `partial_fit` for Keras. Requiring `Pipeline.partial_fit` needs more thought; see https://github.com/scikit-learn/scikit-learn/issues/3299 for more discussion.